### PR TITLE
Docs: How to organize data - Group path

### LINKED ISSUE #3997: Docs: How to organize data - Group path
This will describe the new functionality and already has an open PR #3952

### PR DESCRIPTION
**Pull request by chrisjsewell**
closes #3997 

This is a rebase of #3952 to the `docs_revamp` branch

### CODE REVIEW

File: docs/source/howto/data.rst [hunk 1]
```diff
@@ -514,7 +514,149 @@ To learn about more advanced queries, please see :ref:`the corresponding topics
...
+
+Organise groups in hierarchies
+------------------------------
+
+.. meta::
+   :description: The reStructuredText plaintext markup language
+   :keywords: grouppath
+
+Groups in AiiDA are inherently "flat", in that groups may only contain nodes and not other groups.
+However, the `GroupPath` utility allows one to construct *virtual* group hierarchies based on delimited group labels.
```
Comment (csadorf) [inline docs/source/howto/data.rst:528]:
```suggestion
However it is possible to construct *virtual* group hierarchies based on delimited group labels using the `GroupPath` utility.
```

File: docs/source/howto/data.rst [hunk 2]
```diff
@@ -514,7 +514,149 @@ To learn about more advanced queries, please see :ref:`the corresponding topics
...
+
+`GroupPath` is designed to work in much the same way as python's `pathlib.Path`, whereby paths are denoted by forward slash characters '/' in group labels.
```
Comment (csadorf) [inline docs/source/howto/data.rst:530]:
```suggestion
`GroupPath` is designed to work in much the same way as Python's `pathlib.Path`, whereby paths are denoted by forward slash characters '/' in group labels.
```

File: docs/source/howto/data.rst [hunk 3]
```diff
@@ -514,7 +514,149 @@ To learn about more advanced queries, please see :ref:`the corresponding topics
...
+
+    In [22]: from aiida.orm import UpfFamily
+    In [23]: path2 = GroupPath(cls=UpfFamily)
+    In [24]: path2["base1"].get_or_create_group()
+    Out[24]: (<UpfFamily: "base1" [type core.upf], of user user@email.com>, True)
+
+.. important::
+
+    A `GroupPath` instance will only recognise groups of the instantiated ``cls`` type.
+    The default ``cls`` is ``aiida.orm.Group``:
```
Comment (csadorf) [inline docs/source/howto/data.rst:650]: Can you convert that into an API reference?

File: docs/source/howto/data.rst (hunk 1 above)
Comment (csadorf) [reply docs/source/howto/data.rst:528]: Can you make sure that `GroupPath` links to the API reference?

File: docs/source/howto/data.rst (hunk 2 above)
Comment (csadorf) [reply docs/source/howto/data.rst:530]: Can you link both `GroupPath` and `pahtlib.Path` to the respective API reference docs?

**Comment by csadorf** [comment]
@chrisjsewell Please re-request review when this is ready. Thx.

File: docs/source/howto/data.rst (hunk 3 above)
Comment (chrisjsewell) [reply docs/source/howto/data.rst:651]: fixed in b5f6df7

**Pull request merged**